chapters/02-estrutura-do-programa.md

```diff
@@ -266,43 +266,39 @@ while (number <= 12) {
 
-Então, a variável `number` demonstra o caminho que a variável pode tomar no progresso do programa. Toda hora que o loop se repete, ele é incrementado por 2. Então, no início de toda repetição, ele é comparado com o número 12 para decidir se o programa terminou todo o trabalho que era pretendido a se fazer.
+A variável `number` demonstra uma maneira na qual variáveis podem verificar o progresso de um programa. Toda vez que o loop se repete, `number` é incrementado por `2`. No início de cada repetição, ele é comparado com o número `12` para decidir se o programa terminou de executar todo o trabalho esperado.
 
-Como exemplo de algo realmente útil, podemos agora escrever um programa que calcula e mostra o valor de 2¹⁰ (2 elevado a décima potência). Nós usamos duas variáveis: uma para manter o registro do nosso resultado e uma para contar quantas vezes multiplicamos este resultado por 2. O loop teste se a segunda variável já atingiu 10 e então atualiza ambas variáveis.
-
-```javascript
+Como um exemplo de algo que seja útil, podemos escrever um programa que calcula e mostra o valor de 2¹⁰ (2 elevado à décima potência). Nós usamos duas variáveis: uma para armazenar o resultado e outra para contar quantas vezes multiplicamos esse resultado por 2. O loop testa se a segunda variável já atingiu o valor 10 e então atualiza ambas as variáveis.
 
+```js
 var result = 1;
 var counter = 0;
 while (counter < 10) {
-	result = result * 2;
-	counter = counter + 1;
+  result = result * 2;
+  counter = counter + 1;
 }
 console.log(result);
-// 1024
-
+// → 1024
 ```
 
-O contador pode também iniciar com 1 e checar por `<= 10`, mas, por razões que vamos ver mais a frente, é uma boa ideia usar a contagem iniciando com 0.
+O contador pode também iniciar com `1` e checar o valor com `<= 10`, mas por razões que iremos ver no [Capítulo 4](./04-estruturas-de-dados.md) [TODO:] Adicionar link correto com âncora para a sessão mencionada no texto original: http://eloquentjavascript.net/04_data.html#array_indexing [/TODO], é uma boa ideia se acostumar a usar a contagem iniciando com zero.
 
-Uma estrutura muito similar é o loop `do`. Ele difere somente em um ponto do loop `while`: ele sempre vai executar este a declaração uma vez, e somente então vai iniciar o teste e verificar se precisa pausar. Para demonstrar, o teste é escrito abaixo do corpo do loop:
-
-```javascript
+O loop `do` é uma estrutura de controle similar ao `while`. A única diferença entre eles é que o `do` sempre executa suas declarações ao menos uma vez e inicia o teste para verificar se deve parar ou não apenas após a primeira execução. Para demonstrar isso, o teste aparece após o corpo do loop:
```

Comment:
A escolha por omitir a palavra "loop" é porque ela já está sendo usada para descrever ambas as estruturas no início da frase, ficando implícito que `while` é um loop. Ainda assim, durante a leitura, há várias referências que `while` é um loop para não haver nenhum tipo de confusão.

---

A escolha de não usar uma tradução literal nessa parte é mais para estruturar a frase de modo que facilite a construção e a leitura da frase no português, sem mudar o significado.

---

Essa vírgula está certa.  Ela só não é a melhor escolha para representar o sentido desse caso.  Quando colocamos a vírgula dessa forma (ao redor de "ao menos"), o significado sugere que a execução das declarações **pode** acontecer ou não.  Se não usarmos a vírgula, o significado é de que ela **tem** que acontecer **pelo menos uma vez**, o que é o caso.

Para ficar mais claro, acredito que a frase poderia ser construída assim:
> [...] suas declarações **pelo menos uma vez** e inicia o teste para [...]

O que acha?